lib/s3_meta_sync/syncer.rb

```diff
@@ -244,22 +256,27 @@ def download_content(path)
           "https://s3#{"-#{region}" if region}.amazonaws.com/#{@bucket}/#{path}"
         end
       options = (@config[:ssl_none] ? {:ssl_verify_mode => OpenSSL::SSL::VERIFY_NONE} : {})
-      open(url, options)
-    rescue OpenURI::HTTPError
-      retries ||= 0
-      retries += 1
-      if retries >= 3
+      with_retries(path: path, url: url) { open(url, options) }
+    end
+
+    def with_retries(path:, url:)
+      yield
+    rescue OpenURI::HTTPError, Errno::ECONNRESET => e
+      max_retries = @config[:max_retries] || 2
+      retries ||= Hash.new(0)
+      retries[e.class] += 1
```

Comment:
don't think we need that ... just regular 10 retries should be fine ... usually always  single issue ... also this is per file downloaded and not a global var